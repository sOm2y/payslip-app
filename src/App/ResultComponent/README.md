# the result display component


- should extend when result is calculated (slide out and expand)
- should display the expected result.