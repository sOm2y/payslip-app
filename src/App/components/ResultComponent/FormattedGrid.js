//@flow

import React from 'react';
import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';

class FormattedGrid extends React.PureComponent<{
  value: string | number,
  isTitle?: boolean,
  isMoney?: boolean
}> {
  render() {
    const { value, isTitle, isMoney } = this.props;

    return (
      <Grid item xs={isTitle ? 4 : 8}>
        <Typography type={isTitle ? 'caption' : 'body1'} align={isTitle ? 'left' : 'right'}>
          {isMoney && '$ '}
          {value}
        </Typography>
      </Grid>
    );
  }
}
export default FormattedGrid;
