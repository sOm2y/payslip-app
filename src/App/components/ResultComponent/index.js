//@flow

export { default } from './ResultComponent';
