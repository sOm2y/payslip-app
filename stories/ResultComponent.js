//@flow

import React from 'react';

import ResultComponent from '../src/App/components/ResultComponent';
import { action } from '@storybook/addon-actions';

const result: Payslip = {
  name: 'John Doe',
  payPeriod: '1st March - 31st March',
  grossIncome: 5004,
  incomeTax: 922,
  netIncome: 4082,
  super: 450
};

export default () => <ResultComponent result={result} />;
