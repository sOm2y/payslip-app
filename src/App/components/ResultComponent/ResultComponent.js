//@flow

import React from 'react';

import Typography from 'material-ui/Typography';
import Divider from 'material-ui/Divider';
import Grid from 'material-ui/Grid';
import Container from '../shared/Container';
import Spacer from '../shared/Spacer';

import FormattedGrid from './FormattedGrid';

class ResultComponent extends React.PureComponent<{ result: Payslip }> {
  render() {
    const { result } = this.props;

    return (
      <Container>
        <Typography type="subheading">
          Payslip for <b>{result.name}</b>
        </Typography>
        <Divider />
        <Spacer height={32} />
        <Grid container>
          <FormattedGrid value="Pay Period" isTitle />
          <FormattedGrid value={result.payPeriod} />

          <FormattedGrid value="Gross Income" isTitle />
          <FormattedGrid value={result.grossIncome} isMoney />

          <FormattedGrid value="Income Tax" isTitle />
          <FormattedGrid value={result.incomeTax} isMoney />

          <FormattedGrid value="Net Income" isTitle />
          <FormattedGrid value={result.netIncome} isMoney />

          <FormattedGrid value="Super" isTitle />
          <FormattedGrid value={result.super} isMoney />
        </Grid>
      </Container>
    );
  }
}

export default ResultComponent;
