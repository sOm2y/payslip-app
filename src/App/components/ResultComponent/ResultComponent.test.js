//@flow

import React from 'react';
import ResultComponent from './ResultComponent';
import renderer from 'react-test-renderer';

it('should match snapshot', () => {
  const payslip: Payslip = {
    name: 'John Doe',
    payPeriod: '1st March - 31st March, 2018',
    grossIncome: 10000,
    incomeTax: 2696,
    netIncome: 7304,
    super: 1000
  };

  const tree = renderer.create(<ResultComponent result={payslip} />).toJSON();

  expect(tree).toMatchSnapshot();
});
