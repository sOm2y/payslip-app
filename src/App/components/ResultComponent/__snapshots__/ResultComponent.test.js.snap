// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 Container-container-1"
>
  <h3
    className="MuiTypography-root-29 MuiTypography-subheading-36"
  >
    Payslip for 
    <b>
      John Doe
    </b>
  </h3>
  <hr
    className="MuiDivider-root-53 MuiDivider-default-54"
  />
  <div
    style={
      Object {
        "height": 32,
        "width": undefined,
      }
    }
  />
  <div
    className="MuiGrid-typeContainer-58 MuiGrid-spacing-xs-16-78"
  >
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-4-85"
    >
      <span
        className="MuiTypography-root-29 MuiTypography-caption-39 MuiTypography-alignLeft-41"
      >
        Pay Period
      </span>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-8-89"
    >
      <p
        className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-43"
      >
        1st March - 31st March, 2018
      </p>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-4-85"
    >
      <span
        className="MuiTypography-root-29 MuiTypography-caption-39 MuiTypography-alignLeft-41"
      >
        Gross Income
      </span>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-8-89"
    >
      <p
        className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-43"
      >
        $ 
        10000
      </p>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-4-85"
    >
      <span
        className="MuiTypography-root-29 MuiTypography-caption-39 MuiTypography-alignLeft-41"
      >
        Income Tax
      </span>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-8-89"
    >
      <p
        className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-43"
      >
        $ 
        2696
      </p>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-4-85"
    >
      <span
        className="MuiTypography-root-29 MuiTypography-caption-39 MuiTypography-alignLeft-41"
      >
        Net Income
      </span>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-8-89"
    >
      <p
        className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-43"
      >
        $ 
        7304
      </p>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-4-85"
    >
      <span
        className="MuiTypography-root-29 MuiTypography-caption-39 MuiTypography-alignLeft-41"
      >
        Super
      </span>
    </div>
    <div
      className="MuiGrid-typeItem-59 MuiGrid-grid-xs-8-89"
    >
      <p
        className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-43"
      >
        $ 
        1000
      </p>
    </div>
  </div>
</div>
`;
